proposals/unified-show-diff.md

```diff
@@ -0,0 +1,134 @@
+# Title: Unified show/diff command for all output types
+
+- Enhancement Proposal PR: (leave this empty)
+- Contributors: dberenbaum
+
+# Summary
+
+Establish a unified command to combine and `show` or `diff` all outputs of
+specified types, including:
+
+* Metrics
+* Plots
+* Images (depends on https://github.com/iterative/dvc/discussions/5681)
+
+These would all be put under a single `dvc outputs show/diff` command.
+
+# Motivation
+
+* Enable users to `show` and `diff` multiple different output types within a
+  single command.
+* Reduce number of top-level commands by grouping these together.
+* Promote consistent UI, making it easier for users and for consumption by
+  downstream apps.
+* Generate flexible abstraction for future needs (additional output types or
+  additional functionality across all output types).
+
+# Detailed design
+
+### Definitions
+
+**Outputs**: include any `outs` specified in dvc stages, but only `metrics`,
+`plots`, and `images` will be supported for now. It's possible that other output
+types may be added in the future as needed.
+
+### Command Line Interface
+
+Combine all `show`/`diff` commands into one top-level command:
+
+```console
+usage: dvc outputs [--type] {show,diff}
+
+Commands to visualize and compare outputs (metrics, plots, and images).
+
+
+positional arguments:
+  COMMAND
+    show    Generate report from output files.
+    diff    Compare multiple versions of outputs in a single report.
+
+optional arguments:
+    --type  Restrict report to single output type (metrics, plots, images).
+```
+
+### Requirements of each command:
+
+`show` implementations will:
+* Support file/directory `targets` as positional arguments.
+* By default, show all outputs in the workspace.
+
+`diff` implementations will:
```

Comment:
I'm taking out the parts of the proposal related to iterative/dvc#5693 for now since they are a bit unrelated to the concept of unifying the commands.